compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/Properties.kt

```diff
@@ -17,3 +17,11 @@
 package org.jetbrains.kotlin.cli.common
 
 public val KOTLIN_COMPILER_ENVIRONMENT_KEEPALIVE_PROPERTY = "kotlin.environment.keepalive"
+
+
+fun String?.toBooleanLenient(): Boolean? = when (this?.toLowerCase()) {
```

Comment:
I do not like the suggestion - looks too cumbersome and hiding the logic. But may be it will look better if combined with storing and restoring of the property value